examples/tv-casting-app/linux/main.cpp

```diff
@@ -679,16 +679,19 @@ LinuxCommissionableDataProvider gCommissionableDataProvider;
 int main(int argc, char * argv[])
 {
 #if defined(ENABLE_CHIP_SHELL)
-    Engine::Root().Init();
     std::thread shellThread([]() { Engine::Root().RunMainLoop(); });
-    Shell::RegisterCastingCommands();
 #endif
 
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     SuccessOrExit(err = chip::Platform::MemoryInit());
```

Comment:
The PR summary says that we ensure memory init is called. This however does both memory init and chip stack init and also runs the engine main loop before root is initialized.

It seems the correct thing to do is to do Platform::MemoryInit as the first thing in main, not moving the root init before Root::MainLoop